src/actions/tests/pause.spec.js

```diff
@@ -23,27 +33,68 @@ const mockThreadClient = {
 };
 
 describe("pause", () => {
-  it("should set and clear the command", async () => {
-    const { dispatch, getState } = createStore(mockThreadClient);
-    const mockPauseInfo = {
-      frames: [{ id: 1, scope: [], location: { sourceId: "foo1", line: 4 } }],
-      loadedObjects: [],
-      why: {}
-    };
-
-    await dispatch(actions.paused(mockPauseInfo));
-    dispatch(actions.stepIn());
-    expect(isStepping(getState())).toBeTruthy();
-    await stepInResolve();
-    expect(isStepping(getState())).toBeFalsy();
+  describe("paused", () => {
+    it("should detect when we're paused due to an expression exception", async () => {
+      const { dispatch, getState } = createStore(mockThreadClient);
+      const mockPauseInfo = {
+        frames: [{ id: 1, scope: [], location: { sourceId: "foo1", line: 4 } }],
+        loadedObjects: [],
+        why: { type: "exception" }
+      };
+
+      dispatch(actions.addExpression("foo"));
+      dispatch(actions.evaluateExpressions({ frameId: 2 }));
+      await dispatch(actions.paused(mockPauseInfo));
+      expect(hasWatchExpressionErrored(getState())).toBe(true);
+      evaluateResolve();
+    });
+  });
+
+  describe("stepping", () => {
+    it("should set and clear the command", async () => {
+      const { dispatch, getState } = createStore(mockThreadClient);
+      const mockPauseInfo = {
+        frames: [{ id: 1, scope: [], location: { sourceId: "foo1", line: 4 } }],
+        loadedObjects: [],
+        why: {}
+      };
+
+      await dispatch(actions.paused(mockPauseInfo));
+      dispatch(actions.stepIn());
+      expect(isStepping(getState())).toBeTruthy();
+      await stepInResolve();
+      expect(isStepping(getState())).toBeFalsy();
+    });
   });
 
-  it("should not evaluate expression while stepping", async () => {
-    const client = { evaluate: jest.fn() };
-    const { dispatch } = createStore(client);
+  describe("resumed", () => {
+    it("should not evaluate expression while stepping", async () => {
+      const client = { evaluate: jest.fn() };
+      const { dispatch } = createStore(client);
 
-    dispatch(actions.stepIn());
-    await dispatch(actions.resumed());
-    expect(client.evaluate.mock.calls.length).toEqual(0);
+      dispatch(actions.stepIn());
+      await dispatch(actions.resumed());
+      expect(client.evaluate.mock.calls.length).toEqual(0);
+    });
+
+    it("resuming", async () => {
+      const { dispatch, getState } = createStore(mockThreadClient);
+      const mockPauseInfo = {
+        frames: [{ id: 1, scope: [], location: { sourceId: "foo1", line: 4 } }],
```

Comment:
this `frames` stub is used in multiple tests. maybe you could put it somewhere in a function ? So if/when the structure of a frame changes you don't miss changing the mock somewhere